bson/src/main/org/bson/Renderable.java

```diff
@@ -0,0 +1,20 @@
+package org.bson;
+
+import org.bson.codecs.configuration.CodecRegistry;
+
+/**
+ * An interface for types that are able to render themselves into a {@code BsonDocument}.
+ *
+ * @since 3.0
+ */
+public interface Renderable {
+    /**
+     * Render the filter into a BsonDocument.
+     *
+     * @param documentClass the document class in scope for the collection
+     * @param codecRegistry the codec registry
+     * @param <TDocument> the type of the document class
+     * @return the filter as a BsonDocument
+     */
+    <TDocument> BsonDocument render(Class<TDocument> documentClass, CodecRegistry codecRegistry);
```

Comment:
This may be a problem.  The reason for including it is to allow implementors to re-write filters, projections, sorts, and updates based on knowledge of the class of the document that is in effect for the collection.

Consider a Person class, a MongoCollection<Person>, and a Codec<Person> that transforms the "name" member of Person to an "n" field in the BSON representation.  Then consider a query like this:

     coll.find(eq("name", "Ross"))

when the filter is rendered to a BsonDocument, we'd like to be able to transform the filter to 

     {n : "Ross"}

and in order to do that we need to know about the Person class, so we can look up its Codec in the registry and pull out this information (assuming we have a sub-interface of Codec which provides this information).

The problem is that with a single Renderable interface, we don't know whether we're rendering a filter, a projection, a sort, or an update.